stan/math/rev/mat/fun/sqrt_spd.hpp

```diff
@@ -35,19 +35,23 @@ class sqrt_spd_op {
   //  derivative-or-differential-of-symmetric-square-root-of-a-matrix
 
   template <std::size_t size>
-  std::tuple<Eigen::VectorXd>
+  std::tuple<Eigen::MatrixXd>
   multiply_adjoint_jacobian(const std::array<bool, size> &needs_adj,
-                            const Eigen::VectorXd &adj) const {
+                            const Eigen::MatrixXd &adj) const {
     using Eigen::kroneckerProduct;
     using Eigen::MatrixXd;
+    Eigen::MatrixXd output(K_, K_);
+    Eigen::Map<const Eigen::VectorXd> map_input(adj.data(), adj.size());
+    Eigen::Map<Eigen::VectorXd> map_output(output.data(), output.size());
```

Comment:
Similar, how can `map_output` be a `Map<VectorXd>` but `output` be a square `MatrixXd`?

Reply:
And why is `map_output` necessary? Can we just do the linear algebra to form `output` and do `std::make_tuple(output)`?

Reply:
The maps are just an eigen thing for mapping one data structure onto another. So the math happens in terms of the vectors like Abhishek had it but it just ends up in the right place in the matrices. I think it should be okay in terms of efficiency?